src/socket.rs

```diff
@@ -269,7 +293,7 @@ impl Socket {
             // Cast the address to something the nrf-modem understands
             let address = NrfSockAddr::from(address);
 
-            register_socket_waker(cx.waker().clone(), self.fd);
+            register_socket_waker(cx.waker().clone(), self.fd, self.dir.into());
```

Comment:
Same as in the connection function let's put this at direction TX.
To be clear, this socket cannot TX or RX yet, so it doesn't really matter which we choose.